GoogleForCreators/web-stories-wp
# Shortpixel Image Optimizer Incompatibility

## Context

Plugins like Shortpixel are used for image optimizations within WordPress.

We upload an image for every custom saved page template. These are only used in the editor and should not be optimized by Shortpixel.



## Summary

Adds a filter per

https://github.com/GoogleForCreators/web-stories-wp/issues/10797#issuecomment-1164154035

To check for a page template image and prevent the optimization



## Relevant Technical Choices



## To-do



## User-facing changes



## Testing Instructions






This PR can be tested by following these steps:

Via #10797

1. Install [Shortpixel Image Optimizer plugin](https://wordpress.org/plugins/shortpixel-
2. image-optimiser/)
3. Go to Web Stories > Explore Templates > Use (any) template
4. Go to Page template panel
5. Save page template
6 In Shortpixel stats, see that it cost a credit to optimize the page template image

> Shortpixel should not optimize page template images

Note: Another way to test is to add a media image with `web-stories-page-template` in the name. 

Press the `Optimize now button` -- watch the Network tab (ajax request) + response
![Screen Shot 2022-06-29 at 4 58 48 PM](https://user-images.githubusercontent.com/62242/176543733-10f2d8cd-0598-4431-89aa-92c058f8edc1.png)

## Reviews

### Does this PR have a security-related impact?

No



### Does this PR change what data or activity we track or use?

No



### Does this PR have a legal-related impact?

No



## Checklist




---



Fixes #10797
